src/sc/sidechainTxsCommitmentBuilder.cpp

```diff
@@ -151,15 +151,49 @@ bool SidechainTxsCommitmentBuilder::add_fwt(const CTxForwardTransferOut& ccout,
     const uint160& fwt_mc_return_address = ccout.mcReturnAddress;
     BufferWithSize bws_fwt_return_address((unsigned char*)fwt_mc_return_address.begin(), fwt_mc_return_address.size());
 
-    bool ret = zendoo_commitment_tree_add_fwt(const_cast<commitment_tree_t*>(_cmt),
-         scid_fe,
-         ccout.nValue,
-         &bws_fwt_pk,
-         &bws_fwt_return_address,
-         &bws_tx_hash,
-         out_idx,
-         &ret_code
-    );
+    // Check against the number of sidechains currently in the commitment tree
+    if ((cbg.cbgaMap.size() + cbg.cbgcMap.size()) == cbg.SC_LIMIT) {
+        LogPrint("sc", "%s():%d - scTxsCommitment building failed: too many sidechains, when adding scId[%s].\n",
+            __func__, __LINE__, ccout.GetScId().ToString());
+        return false;
+    }
```

Comment:
Would it be possible that this check fails in the following scenario?
- A transaction with `SC_LIMIT` sidechains
- One _Forward Transfer Output_ for each sidechain
- Then we add a new _Forward Transfer Output_ for an **existing sidechain**

If that's the case, should we perform the check only if the sidechain ID is not already present in the tree?

Reply:
I'd also consider moving from equality check to inequality check (even if I understand fwt are added one by one)

Possibly a check like this:

```
if ((cbg.cbgaMap.size() + cbg.cbgcMap.size()) + (1 - cbg.cbgaMap.count(ccout.GetScId())) > cbg.SC_LIMIT)
```

could handle the scenario described above?

Similar comment applies to `add_bwtr` and `add_cert` (`cbg.cbgaMap`) and `add_csw` (`cbg.cbgcMap`).

Reply:
Yes, the `if` condition was not correct. Addressed in e6db47d33. 
@JackPiri I prefer to state it in this way:
`if ((cbg.cbgaMap.count(ccout.GetScId()) == 0) && ((cbg.cbgaMap.size() + cbg.cbgcMap.size()) == cbg.SC_LIMIT))`
which in my opinion is more clear.

Reply:
Thanks, the current checks seem to be working properly now.

I consider this comment as resolved, I just wanted to propose an additional improvement: given that this check is performed at least 3 or 4 times in different parts of the commitment builder, can we encapsulate it somehow in the class (for instance having functions like `ceasedTreeContains()`, `aliveTreeContains()`, `totalSize()`, `isFull()`, etc.)?

Reply:
I consider this comment as resolved.